tests/conftest.py

```diff
@@ -13,29 +13,27 @@ def tmp_dir(tmp_path_factory):
     tmp_path = tmp_path_factory.getbasetemp() / "image_cache"
     tmp_path.mkdir()
 
-    # download the (only) test images via git's sparse-checkout
-    current_path = os.getcwd()
-    os.chdir(tmp_path)
-    git_version = subprocess.run(
-        ["git", "--version"], stdout=subprocess.PIPE
-    ).stdout.decode("utf-8")[12:18]
-    major, minor, _ = git_version.split(".")
-
-    if int(major) == 2 and int(minor) < 30:
-        # --sparse was introduced in git 2.30.0 (I think)
-        os.system(
-            "git clone --depth 1 --filter=blob:none --no-checkout https://github.com/imageio/imageio-binaries.git ."
-        )
-        os.system("git sparse-checkout init --cone")
-        os.system("git sparse-checkout set test-images")
-    else:
-        # sparse checkout on GH Actions
-        os.system(
-            "git clone --sparse --filter=blob:none https://github.com/imageio/imageio-binaries.git ."
-        )
-        os.system("git sparse-checkout init --cone")
-        os.system("git sparse-checkout set test-images")
-    os.chdir(current_path)
+    if not os.path.isfile(tmp_path / "chelsea.png"):
+        # This function gets called for each test func that uses tmp_dir or image_files.
+        # We only need to download once (per test session)
+
+        # Get list of images. authenticate with our GH token so that the rate limit is per-repo
+        api_endpoint = "https://api.github.com/repos/imageio/imageio-binaries"
+        token = os.getenv("GITHUB_TOKEN")
+        headers = {}
+        if token:
+            headers["Authorization"] = f"token {token}"
+        r = requests.get(api_endpoint + "/contents/test-images", headers=headers)
+        image_info_dicts = r.json()
+        if isinstance(image_info_dicts, dict):
+            image_info_dicts = list(image_info_dicts.values())
+
+        # Download the images
+        for image_info in image_info_dicts:
+            filename = tmp_path / image_info["name"]
+            r = requests.get(image_info["download_url"])
+            with open(filename, "bw") as f:
+                f.write(r.content)
```

Comment:
```suggestion
        # Download the images
        downloader = requests.Session()
        for image_info in image_info_dicts:
            response = downloader.get(image_info["download_url"])
            response.raise_for_status()
            (tmp_path / image_info["name"]).write_bytes(response.content)
```

I would refactor like this. A `Session` will keep the connection alive between request and avoid the overhead of re-establishing a connection for each download. Also, `Path` has a convenience wrappers so we can avoid some boilerplate. Here `write_bytes`.